app/models/version.rb

```diff
@@ -7,6 +7,7 @@ class Version < ActiveRecord::Base
 
   before_save :update_prerelease
   before_validation :full_nameify!
+  after_create :set_info_checksum
```

Comment:
@fotanus I am starting to feel that this is not right place add this. New version object is [created in Puhser](https://github.com/rubygems/rubygems.org/blob/master/app/models/pusher.rb#L85-L88), and is saved when [save! is called on `rubygem`](https://github.com/rubygems/rubygems.org/blob/master/app/models/rubygem.rb#L208):
```Ruby
def update_attributes_from_gem_specification!(version, spec)
    Rubygem.transaction do
      save!  # this is point where info_checksum will be written
      update_versions! version, spec  # this sets `required_rubygem_version` and `required_ruby_version` among other things
      update_dependencies! version, spec # this defines dependencies
      update_linkset! spec
    end
  end
```

As you can see info like `required_rubygem_version`, `required_ruby_version` and `dependencies` are saved after version is created. We use these attributes in `compact_index_info`, which in turn is used by `set_info_checksum`. This means that using it here will generate `info_checksum` with incomplete information.

Thoughts?